src/components/elements/Inputs/textArea/TextArea.tsx

```diff
@@ -1,41 +1,69 @@
 import cn from "classnames";
-import { DetailedHTMLProps, TextareaHTMLAttributes, useId } from "react";
+import { ChangeEvent, DetailedHTMLProps, TextareaHTMLAttributes, useId, useState } from "react";
 import { UseFormReturn } from "react-hook-form";
 
 import InputWrapper, { InputWrapperProps } from "@/components/elements/Inputs/InputElements/InputWrapper";
 
+import { useTextAreaAuto } from "./TextArea.hooks";
+
 export interface TextAreaProps
   extends InputWrapperProps,
     Omit<DetailedHTMLProps<TextareaHTMLAttributes<HTMLTextAreaElement>, HTMLTextAreaElement>, "form"> {
   name: string;
   formHook?: UseFormReturn<any>;
 }
 
-const TextArea = ({ formHook, className, ...inputWrapperProps }: TextAreaProps) => {
-  const { label, description, containerClassName, error, required, feedbackRequired, ...inputProps } =
-    inputWrapperProps;
+const TextArea = ({ formHook, className, onChange: externalOnChange, ...inputWrapperProps }: TextAreaProps) => {
+  const {
+    label,
+    description,
+    containerClassName,
+    error,
+    required,
+    feedbackRequired,
+    labelClassName,
+    labelVariant,
+    ...inputProps
+  } = inputWrapperProps;
   const id = useId();
   const inputClasses = cn(
     "min-h-[150px] w-full rounded-lg px-3 py-2 focus:border-primary-500 outline-none transition-all duration-300 ease-in-out focus:ring-transparent",
-    { className: className },
+    className,
     { "border-light ": !error },
     { ["border border-error focus:border-error"]: error }
   );
   if (error && formHook?.watch(inputWrapperProps.name)) {
     formHook && formHook.trigger();
     formHook && formHook.reset(formHook.getValues());
   }
+  const [textValue, setTextValue] = useState("");
+  const handleTextAreaChange = (event: ChangeEvent<HTMLTextAreaElement>) => {
+    if (externalOnChange) {
+      externalOnChange(event);
+    }
```

Comment:
This can be simplified `externalOnChange?.(event)`. Also, not a big deal for a lightweight callback like this, but it could be wrapped in `useCallback(..., [externalOnChange, setTextValue])`

Reply:
done